spec/forms/two_factor_options_form_spec.rb

```diff
@@ -32,6 +44,12 @@
       end
     end
 
+    it 'includes analytics hash with a methods count of zero' do
+      expect(FormResponse).to receive(:new).with(hash_including(analytics_payload))
+
+      subject.submit(selection: 'sms')
```

Comment:
I think we can have better tests if we check the output, not stubbing (I know a lot of existing tests stub)

```suggestion
      result = subject.submit(selection: 'sms')
      
      expect(result.success?).to eq(true)
      expect(result.to_h).to eq(analytics_payload)
```